langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbTestFixture.java

```diff
@@ -0,0 +1,115 @@
+    private MongoDbEmbeddingStore embeddingStore;
+
+    MongoDbTestFixture(MongoClient client) {
+        this.dbName = "test_database_" + ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+        this.client = client;
+    }
+
+    MongoDbTestFixture initialize() {
+        return this.initialize(b -> b);
+    }
+
+    MongoDbTestFixture initialize(Function<MongoDbEmbeddingStore.Builder, MongoDbEmbeddingStore.Builder> initializer) {
+        this.embeddingStore = initializer.apply(getDefaultMongoDbEmbeddingStoreBuilder()).build();
+        return this;
+    }
+
+    static void assertDoContainerTests() {
+        Assumptions.assumeTrue(getenv("MONGODB_DISABLE_CONTAINER_TESTS") == null);
+    }
+
+    EmbeddingStore<TextSegment> getEmbeddingStore() {
+        return embeddingStore;
+    }
+
+    static MongoClient createClientFromContainer(MongoDBAtlasLocalContainer mongodb) {
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(mongodb.getConnectionString()));
+        return createClientFromBuilder(builder);
+    }
+
+    static MongoClient createClientFromEnv() {
+        boolean hasParts = HOST != null && USERNAME != null && PASSWORD != null;
+        Assumptions.assumeTrue(hasParts || CONNECTION_STRING != null,
+                "Skipping test: host/username/password, or connection string must be provided via env vars.");
+
+        String connectionString = CONNECTION_STRING != null
+                ? CONNECTION_STRING
+                : format("mongodb+srv://%s:%s@%s/?retryWrites=true&w=majority", USERNAME, PASSWORD, HOST);
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(connectionString));
+        return createClientFromBuilder(builder);
+    }
+
+    private static MongoClient createClientFromBuilder(MongoClientSettings.Builder builder) {
+        MongoClientSettings mongoClientSettings = builder
+                .uuidRepresentation(UuidRepresentation.STANDARD)
+                .serverApi(ServerApi.builder().version(ServerApiVersion.V1).build())
+                .build();
+        return MongoClients.create(mongoClientSettings);
+    }
+
+    private MongoDbEmbeddingStore.Builder getDefaultMongoDbEmbeddingStoreBuilder() {
+        IndexMapping indexMapping = IndexMapping.builder()
+                .dimension(EMBEDDING_MODEL.dimension())
+                .metadataFieldNames(Sets.newHashSet("test-key"))
+                .build();
+        return MongoDbEmbeddingStore.builder()
+                .fromClient(client)
+                .databaseName(dbName)
+                .collectionName("test_collection")
+                .indexName("test_index")
+                .indexMapping(indexMapping)
+                .createIndex(true);
+    }
+
+    void afterTests() {
+        getDatabase().drop();
+        client.close();
+    }
```

Comment:
not sure we should drop it. Let's maybe create unique tables for each test. Or reuse test table across tests providing unique identifiers, unique per test case

Reply:
Could you describe the potential issue with dropping the database?

I believe I understand what you mean by unique collections (we would use a random name for a collection instead of the database), but could you explain what you mean by unique identifiers? Does this refer to the test data?

Reply:
I mean, creating and dropping DB for every test or test class might be expensive=>slow. To avoid this, we can use random collection names, or, if we use a single collection, the collection items should be unique per test. We can generate unique ID field in `@BeforeEach` and use it in tests as to isolate data, related to a particular test run

Reply:
If you delete the last collection in a database, you will find the database is no longer present. Dropping the database is not slow.

In the [053d558](https://github.com/langchain4j/langchain4j/pull/1932/commits/053d558ec079dc3587ae38ac2062e454b9389dec) commit, I changed the code to use random collection names. This is not for any performance reason, but because it is possible to configure permissions where dropping the database is stricter than dropping a collection, and we wanted to avoid this config issue.